### PR TITLE
fix(content-releases): Send the populate entry when publish

### PR DESCRIPTION
**Pull request by Feranchz**
### What does it do?

Entity manager needs the complete and populated entry when publishing, so we need to send entries populated to handle validation on components.

### How to test it?

1. Create a release and add an entry with components and these components should have some validation errors
2. Try to publish the release
3. It should return an error

### CODE REVIEW

**Review by simotae14 — COMMENTED** [review]
😮

**Review by markkaylor — APPROVED** [review]
Tested locally and it works, thanks for the quick fix

**Comment by Feranchz** [comment]
I added more code to populate entries we are going to publish manually. In the code I write the explanation but let me know if you have any doubt. Now this is working even for components inside components and all that edge cases (I hope)

File: packages/core/content-releases/server/src/services/release.ts [hunk 1]
```diff
@@ -441,43 +443,80 @@ const createReleaseService = ({ strapi }: { strapi: LoadedStrapi }) => ({
...
+              id: {
+                $in: actionsToPublishIds,
+              },
+            },
+            populate,
+          }
+        )) as Entity[];
+
+        const actionsToUnpublish = (await strapi.entityService.findMany(
+          contentTypeUid as UID.ContentType,
```
Comment (markkaylor) [inline packages/core/content-releases/server/src/services/release.ts:503]: I guess `entriesToPublish` would be a bit more clear. Unfortunate we have to fetch all the entries :( maybe we can find a way to improve it in the future.

File: packages/core/content-releases/server/src/services/release.ts [hunk 2]
```diff
@@ -441,43 +443,80 @@ const createReleaseService = ({ strapi }: { strapi: LoadedStrapi }) => ({
...
     /**
      * We separate publish and unpublish actions group by content type
+     * And we keep only their ids to fetch them later to get all information needed
      */
     const actions: {
       [key: UID.ContentType]: {
-        publish: ReleaseAction['entry'][];
-        unpublish: ReleaseAction['entry'][];
+        actionsToPublishIds: ReleaseAction['entry']['id'][];
+        actionsToUnpublishIds: ReleaseAction['entry']['id'][];
```
Comment (markkaylor) [inline packages/core/content-releases/server/src/services/release.ts:451]: I think we should be careful about the wording. We are publishing entries on the action so I think we should refer to these as `entriesToPublishIds` and I would review the other variable names as well.

**Review by markkaylor — COMMENTED** [review]
The polymorphic relation is feeling less and less necessary but I'm not confident we could get rid of it. 

I tested it locally and it is working even for nested components nice work. One small thing, if I have a nested component, set a field as required on that nested component, and leave the field empty then the Status text in Release will say Ready to publish however it will throw an error if you try to publish.

**Review by markkaylor — APPROVED** [review]
> I tested it locally and it is working even for nested components nice work. One small thing, if I have a nested component, set a field as required on that nested component, and leave the field empty then the Status text in Release will say Ready to publish however it will throw an error if you try to publish.

This seems to be the case for bulk publish as well btw

Looks good tested locally, thanks for the quick fix

**Pull request merged**